web-common/src/features/dashboards/state-managers/loaders/DashboardStateDataLoader.ts

```diff
@@ -201,34 +229,39 @@ export class DashboardStateDataLoader {
 
     this.initExploreState = getCompoundQuery(
       [
-        this.defaultExploreState,
+        this.rillDefaultExploreState,
+        this.exploreStateFromYAMLConfig,
         this.exploreStateFromSessionStorage,
         this.partialExploreStateFromUrlForInitAndErrors,
         ...(bookmarkOrTokenExploreState ? [bookmarkOrTokenExploreState] : []),
       ],
       ([
-        defaultExploreState,
+        rillDefaultExploreState,
+        exploreStateFromYAMLConfig,
         exploreStateFromSessionStorage,
         partialExploreStateFromUrlForInitAndErrors,
         bookmarkOrTokenExploreState,
       ]) => {
-        // type guards. other fields dont need it since we have chaining `??`
-        if (!defaultExploreState) {
+        if (!rillDefaultExploreState || !exploreStateFromYAMLConfig) {
           return undefined;
         }
 
-        const initExploreState = {
-          // Since this is a complete state, we need the complete default explore state which works as a base.
-          ...defaultExploreState,
+        const exploreStateOrder = [
           // 1st priority is the state from session storage.
-          ...(exploreStateFromSessionStorage ??
-            // Next priority is the state loaded from url params. It will be undefined if there are no params.
-            partialExploreStateFromUrlForInitAndErrors?.partialExploreStateFromUrlForInit ??
-            // Next priority is one of the other source defined.
-            // For cloud dashboard it would be home bookmark if present.
-            // For shared url it would be the saved state in token
-            bookmarkOrTokenExploreState),
-        } as MetricsExplorerEntity;
+          exploreStateFromSessionStorage,
+          // Next priority is the state loaded from url params. It will be undefined if there are no params.
+          partialExploreStateFromUrlForInitAndErrors?.partialExploreStateFromUrlForInit,
+          // Next priority is one of the other source defined.
+          // For cloud dashboard it would be home bookmark if present.
+          // For shared url it would be the saved state in token
+          bookmarkOrTokenExploreState,
+          // Next priority is the defaults from yaml config.
+          exploreStateFromYAMLConfig,
+          // Finally the fallback of rill default explore which will have the complete set of config.
+          rillDefaultExploreState,
+        ].filter(Boolean) as Partial<MetricsExplorerEntity>[];
```

Comment:
Because we're merging the `rillDefaultExploreState`, which should be a full `MetricsExplorerEntity` (right?), we can return a full `MetricsExplorerEntity`, not a `Partial`